core/drlibc/drlibc_x86.asm

```diff
@@ -762,51 +762,76 @@ GLOBAL_LABEL(load_dynamo_failure:)
  * signature:
  *   thread_id_t dynamorio_clone(uint flags, byte *newsp, void *ptid, void *tls,
  *                               void *ctid, void (*func)(void))
+ * i#6514: If newsp is NULL then that tells the kernel to give the child the
+ * same value for SP as the parent.
  */
         DECLARE_FUNC(dynamorio_clone)
 GLOBAL_LABEL(dynamorio_clone:)
-        /* save func for use post-syscall on the newsp.
-         * when using clone_record_t we have 4 slots we can clobber.
+        /* Save func for use post-syscall on the newsp.
+         * This is tricky because we have to handle the case of newsp == NULL.
          */
 # ifdef X64
-        sub      ARG2, ARG_SZ
-        mov      [ARG2], ARG6 /* func is now on TOS of newsp */
-        /* all args are already in syscall registers */
+        /* The syscall preserves all registers except rax, rcx, r11. */
+        push     r15
+        mov      r15, ARG6 /* Func is now in r15. */
+        sub      REG_XSP, 2*ARG_SZ /* Keep xsp aligned, for newsp==NULL case. */
```

Comment:
But it's already aligned after the push on line 775.  Seems like this line can be removed.

Reply:
Done